cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,260 @@ See the Apache™ Twill®
+      // Properties for Explore (to create a partitioned Hive table)
+      .setEnableExploreOnCreate(true)
+      .setExploreFormat("csv")
+      .setExploreSchema("date STRING, winner STRING, loser STRING, winnerpoints INT, loserpoints INT")
+      .build());
+
+This results in the creation of an external table in Hive with the schema given in the
+``setExploreSchema()``. The supported format are ``text`` and ``csv``. Both mean that the
+format is text. For ``csv``, the field delimiter is a comma, whereas for ``text``, you can
+specify the field delimiter. For example, to use a colon as the field separator::
+
+      .setExploreFormat("text")
+      .setExploreFormatProperty("delimiter", ":");
+
+If your file format is not text, you can still explore the dataset, but you need to give
+detailed instructions when creating the dataset. For example, to use Avro as the file
+format::
+
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", SCHEMA_STRING)
+
+You need to specify the SerDe, the input format, the output format, and any additional properties
+any of these may need as table properties. This is an experimental feature and only tested for
+Avro; see the :ref:`StreamConversion <examples-stream-conversion>` example for more details.
+
+======================
+TimePartitionedFileSet
+======================
+
+TimePartitionedFileSets are a special case (and in fact, a subclass) of PartitionedFileSets, where
+the partitioning is fixed to five integers representing the year, month, day of the month, hour of the day,
+and minute of a partition's time. For convenience, it offers methods to address the partitions by
+time instead of partition key or filter. The time is interpreted as milliseconds since the Epoch.
+
+The following convenience methods provide access to partitions by time instead of a partition key::
+
+  @Nullable
+  public TimePartition getPartitionByTime(long time);
+
+  public Set<TimePartition> getPartitionsByTime(long startTime, long endTime);
+
+  @Nullable
+  public TimePartitionOutput getPartitionOutput(long time);
+
+Essentially, these methods behave the same as if you convert the time arguments into partition keys and
+call the corresponding methods of ``PartitionedFileSet`` with the resulting partition keys. In addition:
+
+- The returned partitions have an extra method to retrieve the partition time as long.
+- The start and end time of ``getPartitionsByTime()`` do not correspond directly to a single partition filter,
+  but to a series of partition filters. For example, to retrieve the partitions between November 2014 and
+  March 2015, you need two partition filters: One for the months of November to December of 2014, and one
```

Comment:
filters: one for the months of November through December of 2014, and one
for January through March of 2015.